src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -1622,8 +1554,17 @@ public boolean process(Person person, long time) {
 
     public ObservationGroup clone() {
       ObservationGroup clone = (ObservationGroup) super.clone();
-      clone.codes = codes;
-      clone.observations = observations;
+      
+      // IMPORTANT: because each observation gets process()ed when the state gets processed,
+      // we need to ensure we deep clone the list
+      // (otherwise as this gets passed around, the same objects are used for different patients
+      // which causes weird and unexpected results)
+      List<Observation> cloneObs = new ArrayList<>(observations);
```

Comment:
Should this be new ArrayList<>(observations.size()) instead?

Reply:
Yeah I think I had chosen the complicated way to implement deep copying. Updated